packages/web3/src/signer/types.test.ts

```diff
@@ -0,0 +1,252 @@
+      }
+
+      expect(signerAddress.signerAddress).toBe('0x456')
+      expect(signerAddress.signerKeyType).toBeUndefined()
+    })
+  })
+
+  describe('Transaction Params', () => {
+    describe('SignTransferTxParams', () => {
+      it('should create a valid SignTransferTxParams object', () => {
+        const params: SignTransferTxParams = {
+          signerAddress: '0x123',
+          destinations: [
+            {
+              address: '0x456',
+              attoAlphAmount: '1000000000000000000'
+            }
+          ],
+          signerKeyType: 'default',
+          gasAmount: 100,
+          gasPrice: '1000000000000000'
+        }
+
+        expect(params.signerAddress).toBe('0x123')
+        expect(params.destinations).toHaveLength(1)
+        expect(params.signerKeyType).toBe('default')
+        expect(params.gasAmount).toBe(100)
+        expect(params.gasPrice).toBe('1000000000000000')
+      })
+    })
+
+    describe('SignDeployContractTxParams', () => {
+      it('should create a valid SignDeployContractTxParams object', () => {
+        const params: SignDeployContractTxParams = {
+          signerAddress: '0x123',
+          bytecode: '0xabcdef',
+          signerKeyType: 'bip340-schnorr',
+          initialAttoAlphAmount: '1000000000000000000',
+          initialTokenAmounts: [{ id: 'token1', amount: '100' }],
+          issueTokenAmount: '500',
+          issueTokenTo: '0x456',
+          gasAmount: 200,
+          gasPrice: '2000000000000000'
+        }
+
+        expect(params.signerAddress).toBe('0x123')
+        expect(params.bytecode).toBe('0xabcdef')
+        expect(params.signerKeyType).toBe('bip340-schnorr')
+        expect(params.initialAttoAlphAmount).toBe('1000000000000000000')
+        expect(params.initialTokenAmounts).toHaveLength(1)
+        expect(params.issueTokenAmount).toBe('500')
+        expect(params.issueTokenTo).toBe('0x456')
+        expect(params.gasAmount).toBe(200)
+        expect(params.gasPrice).toBe('2000000000000000')
+      })
+    })
+
+    describe('SignExecuteScriptTxParams', () => {
+      it('should create a valid SignExecuteScriptTxParams object', () => {
+        const params: SignExecuteScriptTxParams = {
+          signerAddress: '0x123',
+          bytecode: '0xabcdef',
+          signerKeyType: 'default',
+          attoAlphAmount: '1000000000000000000',
+          tokens: [{ id: 'token1', amount: '100' }],
+          gasAmount: 150,
+          gasPrice: '1500000000000000',
+          gasEstimationMultiplier: 1.5
+        }
+
+        expect(params.signerAddress).toBe('0x123')
+        expect(params.bytecode).toBe('0xabcdef')
+        expect(params.signerKeyType).toBe('default')
+        expect(params.attoAlphAmount).toBe('1000000000000000000')
+        expect(params.tokens).toHaveLength(1)
+        expect(params.gasAmount).toBe(150)
+        expect(params.gasPrice).toBe('1500000000000000')
+        expect(params.gasEstimationMultiplier).toBe(1.5)
+      })
+    })
+  })
```

Comment:
Most of these just test general javascript, it would just increase code coverage and maintenance costs I think.

Reply:
@h0ngcha0 okay will sort out the test cases